src/userobjects/BoundingBoxInitializer.C

```diff
@@ -0,0 +1,179 @@
+  {
+    mooseError("Params 'y1' and 'y2' should not be set in 1D simulations");
+  }
+
+  if (_fe_problem.mesh().dimension() != 3 && (isParamSetByUser("z1") || isParamSetByUser("z2")))
+  {
+    mooseError("Params 'z1' and 'z2' should not be set unless you are performing a 3D simulation");
+  }
+
+  std::vector<std::string> bottom_left_stirngs = {"x1", "y1", "z1"};
+  std::vector<std::string> top_right_stirngs = {"x2", "y2", "z2"};
+  for (const auto i : make_range(_mesh_dimension))
+  {
+    if (_top_right(i) <= _bottom_left(i))
+      paramError(top_right_stirngs[i],
+                 top_right_stirngs[i] + " must be greater than " + bottom_left_stirngs[i]);
+  }
+}
+
+std::vector<InitialParticleData>
+BoundingBoxInitializer::getParticleData() const
+{
+
+  // first collect the elements that we will place particles in
+  std::vector<const Elem *> valid_elems;
+
+  for (const auto elem : *_fe_problem.mesh().getActiveLocalElementRange())
+  {
+    unsigned int intersection_count = 0;
+    unsigned int nodes_within = 0;
+    bool elem_added = false;
+    // if one of the nodes is in our box or on the edge
+    // of the box then we need to add this to the elements we want
+    // to place particles in
+    for (const auto & node : elem->node_ref_range())
+    {
+      unsigned int dim_valid = 0;
+      // first we can check to see if one node is within the bounding box
+      // this is sufficient for 1D
+      for (const auto i : make_range(_mesh_dimension))
+      {
+        if (node(i) >= _bottom_left(i) && node(i) <= _top_right(i))
+          dim_valid++;
+      }
+
+      if (dim_valid == _mesh_dimension)
+      {
+        valid_elems.push_back(elem);
+        elem_added = true;
+        break;
+      }
+
+      // if we are in a higher dimension then we need to check for
+      // elements which intersect the bounding box but do not have
+      // any nodes within the bounding box
+      if (_mesh_dimension > 1)
+      {
+        // checking to see which side of each plane each node is on
+        for (const auto i : make_range(2 * _mesh_dimension))
+        {
+          Real plane_value = 0;
+          for (const auto j : make_range(_mesh_dimension))
+          {
+            plane_value += node(j) * _planes[i][j];
+          }
+          plane_value += _planes[i][3];
+
+          if (plane_value >= 0)
+            nodes_within++;
+        }
+        if (nodes_within > 1)
+        {
+          intersection_count++;
+        }
+      }
+    }
+    if (elem_added)
+      continue;
+    // if more than two planes are intersecting the
+    // element then we need to put some particles within it as well
+    if (intersection_count > 1)
+      valid_elems.push_back(elem);
+  }
+
+  // if there are no elements for this processor: do nothing
+  if (valid_elems.size() == 0)
+    return {};
+
+  std::vector<InitialParticleData> data = std::vector<InitialParticleData>();
```

Comment:
```suggestion
  std::vector<InitialParticleData> data;
```

The default constructor for std::vector is just an empty vector, so the RHS is redundant